src/render.rs

```diff
@@ -0,0 +1,301 @@
+            let pos = entity.get_position();
+            match entity {
+                Entity::Player(_) => {
+                    let mut glyph_buffer = glyph_buffer.clone();
+                    let player_sprite = get_player_sprite_4();
+                    for (yi, row) in player_sprite.chars.rows().into_iter().enumerate()
+                    {
+                        for (xi, pixel) in row.iter().enumerate() {
+                            let index = [yi + (pos.y * 2), xi + (pos.x * 4)];
+                            glyph_buffer[index].glyph = pixel.char;
+                            if pixel.fg.is_some() {
+                                glyph_buffer[index].fg = pixel.fg;
+                            }
+                            if pixel.bg.is_some() {
+                                glyph_buffer[index].bg = pixel.bg;
+                            }
+                        }
+                    }
+                    glyph_buffer
+                }
+                entity => draw_square(entity.color(), pos, 4, glyph_buffer),
+            }
+        }
+    }
+}
+
+/// Generates a `RenderGraph` from the provided `World` struct.
+///
+/// # Parameters
+///
+/// - `world`: A reference to a `World` struct, which contains the game's state, including the board and entities.
+///
+/// # Returns
+///
+/// A `RenderGraph` struct that represents the entire scene to be rendered. The root node of the graph corresponds to
+/// the game's board, and its children correspond to the entities present in the world.
+///
+/// # Details
+///
+/// This function creates a hierarchical `RenderGraph` that represents the current state of the game world. The root
+/// node of the graph contains a `RenderItem::Board`, which represents the game board. The root node's children are
+/// `RenderNode`s that each contain a `RenderItem::Entity`, representing the entities in the world. Each entity is
+/// cloned from the `world` and encapsulated in a `RenderNode` without further children (i.e., `children` is `None`).
+///
+/// This graph structure allows for organized and flexible rendering of the game world. By constructing the scene
+/// as a tree of `RenderNode`s, it becomes easier to traverse and render the scene in a systematic way, ensuring
+/// that the board is rendered first, followed by the entities in the correct order.
+///
+/// This function is typically called before rendering to prepare the data needed to generate a visual representation
+/// of the game state.
+///
+/// # Examples
+///
+/// ```rust
+/// let world = World::new(); // Assume World::new initializes a game world
+/// let render_graph = generate_render_graph(&world);
+///
+/// // `render_graph` can now be passed to a rendering function
+/// let area = Rect::new(0, 0, 80, 24); // Example rendering area
+/// let glyph_buffer = glypherize_graph(render_graph, area, render_fn);
+/// ```
+fn generate_render_graph(world: &World) -> RenderGraph {
+    let children = Some(
```

Comment:
I might move the `Some` to just when it's passed to the `RenderNode`. Just because it's in that context that we know Children is optional. But within this space we don't really care.